src/GraphQL/Internal/AST.hs

```diff
@@ -76,12 +130,12 @@ data OperationDefinition = Query    { getNode :: Node }
                          | Mutation { getNode :: Node }
                            deriving (Eq,Show)
 
-data Node = Node Name [VariableDefinition] [Directive] SelectionSet
+data Node = Node (Maybe Name) [VariableDefinition] [Directive] SelectionSet
             deriving (Eq,Show)
 
 -- XXX: Lots of things have names. Maybe we should define a typeclass for
```

Comment:
To be clear, are you suggesting that I remove the `HasName` type class and all of its instances and replace them with specifically named functions that each do the job at hand?

Even golang supports this level of genericness.